taca/nanopore/ONT_run_classes.py

```diff
@@ -475,31 +496,46 @@ def run_anglerfish(self):
             f"--run_name {anglerfish_run_name}",
             f"--threads {n_threads}",
             "--lenient",
-            "--ont_barcodes",
             "--skip_demux",
         ]
 
+        # Make dir to trace Anglerfish run
+        os.mkdir(taca_anglerfish_run_dir)
+        # Copy samplesheet used
+        shutil.copy(self.anglerfish_samplesheet, f"{taca_anglerfish_run_dir}/")
+        # Create files to dump subprocess std
+        stderr_relpath = f"{taca_anglerfish_run_dir}/stderr.txt"
+
+        if self.has_barcode_dirs():
+            anglerfish_command.append("--barcoding")
+
         full_command = [
             # Dump subprocess PID into 'run-ongoing'-indicator file.
             f"echo $$ > {self.anglerfish_ongoing_abspath}",
             # Run Anglerfish in its own environment
             "conda run -n anglerfish " + " ".join(anglerfish_command),
             # Dump Anglerfish exit code into file
             f"echo $? > {self.anglerfish_done_abspath}",
-            # Copy the Anglerfish samplesheet used to start the run into the run dir, for traceability
-            # (The correct anglerfish run dir is identified by it being younger than the "run-ongoing" file)
-            f"new_runs=$(find . -type d -name 'anglerfish_run*' -newer {self.anglerfish_ongoing_abspath})",
-            f"if [[ $(echo '${{new_runs}}' | wc -l) -eq 1 ]] ; then cp {self.anglerfish_samplesheet} ${{new_runs}}/ ; fi",
-            # Regardless of exit status: Remove 'run-ongoing' file.
+            # Move run to subdir
+            f'find {self.run_abspath} -name "anglerfish_run*" -type d -newer {self.run_abspath}/.anglerfish_ongoing '
+            + '-exec mv \{\} '
+            + f'{self.run_abspath}/{taca_anglerfish_run_dir}/ \; '
+            + '-quit',
+            # Remove 'run-ongoing' file.
             f"rm {self.anglerfish_ongoing_abspath}",
         ]
 
+        with open(f"{taca_anglerfish_run_dir}/command.sh", "w") as stream:
+            stream.write("\n".join(full_command))
+
         # Start Anglerfish subprocess
-        process = subprocess.Popen(
-            "; ".join(full_command),
-            shell=True,
-            cwd=self.run_abspath,
-        )
+        with open(stderr_relpath, 'w') as stderr:
+            process = subprocess.Popen(
+                "; ".join(full_command),
```

Comment:
Could we even run the command.sh file?

Reply:
[0baa97c](https://github.com/SciLifeLab/TACA/pull/409/commits/0baa97c9898eb71061b8b2ac2b86b44e52de2c20)